xla/service/gpu/runtime/gpublas_lt_matmul.h

```diff
@@ -35,7 +35,12 @@ void PopulateCublasLtMatmulAttrEncoding(
 // Registers XLA Gpu runtime cuBLASLt custom calls.
 void RegisterMatmulCustomCalls(runtime::DirectCustomCallRegistry& registry);
 
-using MatmulPlanVec = std::vector< se::gpu::BlasLt::MatmulPlanPtr >;
+struct MatmulPlanVec : public std::vector< se::gpu::BlasLt::MatmulPlanPtr > {
```

Comment:
I think that it would be better if this encapsulated the synchronization and didn't derive from a standard container.

```
class MatmulPlanVec final {
public:
absl::StatusOr<se::gpu::BlasLt::MatmulPlanPtr> GetOrCreate(int device_id, std::function<absl::StatusOr<MatmulPlanPtr>> create_plan){
// Hold mutex, do lookup, etc.
}
private:
std::vector<se::gpu::BlasLt::MatmulPlanPtr> plans_;
// unique_ptr is usually not needed. But maybe in your case?
absl::Mutex mu_;
};
```

Reply:
I have refactored the implementation as you suggested: I have only put the body of GetOrCreate into the source file: wanted to keep the header clean.. anyway there is no template code there. Concerning the mutex: yes, I have tried first without the unique pointer but the problem is, State::GetOrCreate() template function requires an object being constructed to be moveable which is not the case for absl::Mutex

Reply:
makes sense